### PR TITLE
compose: Add types to `useKeyboardShortcut`

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Adds/corrects types to `useKeyboardShortcut`. There were pre-existing types but they weren't being type-checked.

Part of #18838

## How has this been tested?
Type checks pass. No runtime changes that should affect anything except for an extra null check.

## Types of changes
New feature.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [N/A] My code follows the accessibility standards. 
- [N/A] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/compose/README.md [hunk 1]
```diff
@@ -291,10 +291,14 @@ throws a warning when using useLayoutEffect in that environment.
...
 
+_Related_
+
+-   <https://craig.is/killing/mice#api.bind> for information about the `callback` parameter.
+
 _Parameters_
 
 -   _shortcuts_ `string[]|string`: Keyboard Shortcuts.
--   _callback_ `Function`: Shortcut callback.
+-   _callback_ `import('mousetrap').MousetrapInstance['bind']`: Shortcut callback.
```
Comment (ellatrix) [inline packages/compose/README.md]: Isn't this just a DOM event callback?

Comment (sarayourfriend) [reply packages/compose/README.md]: It's not quite, from the types:

```js
        bind(
            keys: string | string[],
            callback: (e: ExtendedKeyboardEvent, combo: string) => any,
            action?: string,
        ): this;
```

**Review by tyxla — APPROVED** [review]
Looks great to me 👍

File: packages/compose/README.md (hunk 1 above)
Comment (ciampo) [reply packages/compose/README.md]: Shouldn't the callback be of type `(e: ExtendedKeyboardEvent, combo: string) => any` ?

Comment (sarayourfriend) [reply packages/compose/README.md]: Oh shoot you're right, the callback is passed as the second param to `bind`, my bad. I'll fix this.

File: packages/compose/src/hooks/use-keyboard-shortcut/index.js [hunk 2]
```diff
@@ -37,14 +37,18 @@ function isAppleOS( _window = window ) {
...
 /**
  * Attach a keyboard shortcut handler.
  *
- * @param {string[]|string}         shortcuts  Keyboard Shortcuts.
- * @param {Function}                callback   Shortcut callback.
- * @param {WPKeyboardShortcutConfig} options    Shortcut options.
+ * @see https://craig.is/killing/mice#api.bind for information about the `callback` parameter.
+ *
+ * @param {string[]|string} shortcuts Keyboard Shortcuts.
+ * @param {import('mousetrap').MousetrapInstance['bind']} callback Shortcut callback.
```
Comment (ciampo) [inline packages/compose/src/hooks/use-keyboard-shortcut/index.js]: Following [this comment](https://github.com/WordPress/gutenberg/pull/32168/commits/44f9a5012f9f6ce3044b17a9db6d2a191df9957a#r638647921), shouldn't this be `(e: ExtendedKeyboardEvent, combo: string) => any` too?

Comment (ciampo) [reply packages/compose/src/hooks/use-keyboard-shortcut/index.js]: Nevermind, somehow GitHub didn't load the most recent version of the PR 🤷

**Review by ciampo — APPROVED** [review]
LGTM 🚀

**Pull request merged**